pkg/controller/controlplane/valuesprovider.go

```diff
@@ -453,8 +458,8 @@ func (vp *valuesProvider) getControlPlaneChartValues(
 		gcp.CSIControllerName:          csi,
 		gcp.CSIFilestoreControllerName: csiFilestore,
 		gcp.IngressGCEName: map[string]interface{}{
-			"enabled":  isDualstackEnabled(cluster.Shoot.Spec.Networking, nil),
-			"replicas": extensionscontroller.GetControlPlaneReplicas(cluster, scaledDown, 1),
+			"enabled":  isDualstackEnabled(cluster.Shoot.Spec.Networking, cluster.Shoot.Status.Networking),
```

Comment:
I don't think this will work as expected. When the chart is disabled (`!enabled`) then it won't be deployed at all. That includes the replica change.

In this case it should always be enabled in all cases, pretty much like aws-custom-route/ipam and azure-remedy controllers

Reply:
The chart will not be disabled, when migrating back to single stack. `isDualstackEnabled` will return `true` in this case, as we have still have the two CIDRs in the networking status. Probably, the function Name `isDualstackEnabled` is misleading.